src/hooks/useStage.js

```diff
@@ -1,9 +1,32 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import { createStage } from '../utils/createStage';
 
-const useStage = () => {
+const useStage = (player, resetPLayer) => {
   const [stage, setStage] = useState(createStage());
 
+  useEffect(() => {
+    const updateStage = (prevStage) => {
+      // Clear the stage from previous render.
+      const newStage = prevStage.map((row) =>
+        row.map((oneCell) => (oneCell[1] === 'clear' ? [0, 'clear'] : oneCell))
+      );
+
+      // Draw the new stage(tetromino)
+      player.tetromino.forEach((row, y) => {
+        row.forEach((value, x) => {
+          if (value !== 0) {
+            newStage[y + player.pos.y][x + player.pos.x] = [
+              value,
+              `${player.collided ? 'merged' : 'clear'}`,
+            ];
+          }
+        });
+      });
```

Comment:
I think you need to make condition here if player is collided or not
if yes you will run resetPlayer function
```js
if (player.collided) {
        resetPlayer();
      }
```